kinisi/yeh_hummer.py

```diff
@@ -0,0 +1,328 @@
+"""
+Implementation of the Yeh-Hummer finite-size correction for diffusion coefficients
+from molecular dynamics simulations with periodic boundary conditions.
+
+Based on: Yeh & Hummer, J. Phys. Chem. B 2004, 108, 15873-15879
+"""
+
+# Copyright (c) kinisi developers.
+# Distributed under the terms of the MIT License
+# author: Fabian Zills (pythonfz)
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipp as sc
+from scipy.optimize import curve_fit
+
+from kinisi import __version__
+from kinisi.fitting import FittingBase
+
+from .due import Doi, due
+
+
+def yeh_hummer_linear(inv_L, D_0, slope):
+    """
+    Linear form of Yeh-Hummer equation for fitting.
+
+    D_PBC = D_0 - slope * (1/L)
+
+    where slope = (k_B * T * xi) / (6 * pi * eta)
+
+    :param inv_L: Inverse box lengths (1/L)
+    :param D_0: Infinite-system diffusion coefficient
+    :param slope: Slope containing viscosity information
+    :return: D_PBC values
+    """
+    return D_0 - slope * inv_L
+
+
+@due.dcite(
+    Doi('10.1021/jp0477147'),
+    path='kinisi.yeh_hummer.YehHummer',
+    description='Yeh-Hummer finite-size correction.',
+    version=__version__,
+)
+class YehHummer(FittingBase):
+    """
+    Apply Yeh-Hummer finite-size corrections to diffusion coefficients from MD simulations
+    with periodic boundary conditions.
+
+    The Yeh-Hummer correction formula is:
+    D_PBC = D_0 - (k_B * T * xi) / (6 * pi * eta * L)
+
+    :param diffusion: sc.DataArray with diffusion coefficients and box_length coordinate
+    :param temperature: Temperature (will be extracted from coords if not provided)
+    :param bounds: Optional bounds for [D_0, viscosity] parameters
+    """
+
+    def __init__(self, diffusion, temperature=None, bounds=None):
+        self.diffusion = diffusion
+
+        # Extract box lengths from coordinates
+        self.box_lengths = diffusion.coords['box_length']
+
+        # Handle temperature
+        self.temperature = temperature if isinstance(temperature, sc.Variable) else sc.scalar(temperature, unit='K')
+
+        # Constants
+        self.xi_cubic = 2.837297  # Ewald constant for cubic boxes
```

Comment:
Is this dimensionless?

Reply:
From https://pubs.acs.org/doi/full/10.1021/jp951011v I think the units are `1/L`.